chain/client/src/client.rs

```diff
@@ -1253,9 +1246,11 @@ impl Client {
         apply_chunks_done_callback: DoneApplyChunkCallback,
     ) {
         let chunk_header = partial_chunk.cloned_header();
+        self.chain.blocks_delay_tracker.mark_chunk_completed(&chunk_header, Clock::utc());
+        self.block_production_info
```

Comment:
Note: I think this is a bug fix. It used to be called whenever we receive a PartialEncodedChunk, but that's not the same as collecting the chunk for block production purposes.